clients/drcachesim/tracer/output.cpp

```diff
@@ -752,8 +751,23 @@ output_buffer(void *drcontext, per_thread_t *data, byte *buf_base, byte *buf_ptr
                     DR_ASSERT(is_ok_to_split_before(
                         instru->get_entry_type(pipe_start + header_size),
                         instru->get_entry_size(pipe_start + header_size)));
-                    pipe_start = atomic_pipe_write(drcontext, pipe_start, pipe_end,
-                                                   get_local_window(data));
+                    // Check if we went over the edge waiting for enough entries to
+                    // write. If we did, we simply write till the last ok-to-split ref.
+                    if (mem_ref - pipe_start > ipc_pipe.get_atomic_write_size()) {
+                        // If the following assert triggers, we found too many entries
+                        // without an ok-to-split point.
```

Comment:
Instead of this comment this can be added to the assert with DR_ASSERT_MSG (or `&& "string"` if that weren't available): "found too many entries with no split point" or something.  Then it will be visible to the dev/user if this fires.

Reply:
Done.